File: Older/NFFramework/NFFramework/NFFramework/Demo/MenuViewController.m
//
//  MenuViewController.m
//  NFFramework
//
//  Created by ras-fueled on 25/08/2014.
//  Copyright (c) 2014 Nickelfox. All rights reserved.
//

#import "MenuViewController.h"
#import "ValidationFieldDemoViewController.h"

#define kCellID @"kMenuCellID"

@interface MenuViewController ()<UITableViewDataSource, UITableViewDelegate>
@property(nonatomic, weak)IBOutlet UITableView *tableView;
@end

@implementation MenuViewController

- (void)viewDidLoad {
  [super viewDidLoad];
  [self setupTableView];
}

- (void)didReceiveMemoryWarning {
  [super didReceiveMemoryWarning];
}

- (void)setupTableView {
  self.tableView.delegate = self;
  self.tableView.dataSource = self;
  [self.tableView registerClass:[UITableViewCell class] forCellReuseIdentifier:kCellID];
  [self.tableView reloadData];
}

#pragma mark - UITableViewDataSource, UITableViewDelegate

- (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section {
  return 2;
}

- (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath {
  UITableViewCell *cell = [tableView dequeueReusableCellWithIdentifier:kCellID
                                                          forIndexPath:indexPath];
  cell.accessoryType = UITableViewCellAccessoryDisclosureIndicator;
  cell.textLabel.text = [self textForCellAtIndexPath:indexPath];
  return cell;
}

- (void)tableView:(UITableView *)tableView didSelectRowAtIndexPath:(NSIndexPath *)indexPath {
  [tableView deselectRowAtIndexPath:indexPath animated:YES];
  [self handleRowSelectedAtIndexPath:indexPath];
}

#pragma mark - Helper
- (NSString *)textForCellAtIndexPath:(NSIndexPath *)indexPath {
  NSString *text;
  switch (indexPath.row) {
    case 0:
      text = @"Validation Field";
      break;
    case 1:
      text = @"Connect";
      break;
    default:
      break;
  }
  return text;
}

- (void)handleRowSelectedAtIndexPath:(NSIndexPath *)indexPath {
  switch (indexPath.row) {
    case 0:{ // Validation Field
      [self showDemoForValidationField];
    }
      break;
    case 1:{ // Connect Flow
      [self showDemoForConnectFlow];
    }
      break;
    default:
      break;
  }
}

#pragma mark - Demos
- (void)showDemoForValidationField {
  ValidationFieldDemoViewController *validationFieldVC = [[ValidationFieldDemoViewController alloc] init];
  [self.navigationController pushViewController:validationFieldVC animated:YES];
}

- (void)showDemoForConnectFlow {
  UIStoryboard *storyboard = [UIStoryboard storyboardWithName:@"Connect" bundle:[NSBundle mainBundle]];
  UINavigationController *connectNav = [storyboard instantiateInitialViewController];
  [self.navigationController presentViewController:connectNav animated:YES completion:nil];
}
@end

File: FormValidations/FormValidations/MenuViewController.swift
//
//  MenuViewController.swift
//  FormValidations
//
//  Created by Nickelfox on 04/04/17.
//  Copyright © 2017 Nickelfox. All rights reserved.
//

import UIKit

class MenuViewController: UIViewController {

    @IBOutlet var tableView: UITableView!
    var cellId = "menuCell"
    var nibId = "MenuViewController"
    
    override func viewDidLoad() {
        super.viewDidLoad()
        setupTableView()
    }
    
    func setupTableView() {
        displayView()
        tableView.dataSource = self
        tableView.delegate = self
        tableView.register(UITableViewCell.self, forCellReuseIdentifier: cellId)
        tableView.reloadData()
    }
    
    func displayView() {
        let menuView = Bundle.main.loadNibNamed(nibId, owner: self, options: nil)?.first as! UIView
        self.view = menuView
    }
}

extension MenuViewController: UITableViewDataSource, UITableViewDelegate {
    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
        return 2
    }
    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
        let cell = tableView.dequeueReusableCell(withIdentifier: cellId, for: indexPath) as UITableViewCell
        cell.accessoryType = .disclosureIndicator
        cell.textLabel?.text = textForCell(at: indexPath)
        return cell
    }
    
    func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
        tableView.deselectRow(at: indexPath, animated: true)
        handleRowSelected(at: indexPath)
    }
    
    //MARK: Helper Methods
    
    func textForCell(at indexPath: IndexPath) -> String{
        var text: String = ""
        switch indexPath.row {
        case 0:
            text = "Validation Field"
        case 1:
            text = "Connect"
        default:
            break
        }
      return text
    }
    
    func handleRowSelected(at indexPath: IndexPath) {
        switch indexPath.row {
        case 0: //Validation Field
            showValidationFieldDemo()
        case 1: // Connect
            showConnectDemo()
        default:
            break
        }
    }
    
    func showValidationFieldDemo() {
        let validationFieldVC = ValidationFieldDemoViewController()
        navigationController?.pushViewController(validationFieldVC, animated: true)
    }
    
    func showConnectDemo() {
        let storyboard = UIStoryboard(name: "Main", bundle: Bundle.main)
        let connectVC = storyboard.instantiateInitialViewController()
        navigationController?.present(connectVC!, animated: true, completion: nil)
        
    }
    
}
